projects/file-parsing/fileparser/jsonparser.go

```diff
@@ -0,0 +1,28 @@
+package jsonparser
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+)
+
+type Player struct {
+	Name      string `json:"name"`
+	HighScore int    `json:"high_score"`
+}
+
+type Players []Player
+
+func ParseJSON(filename string) (players Players, err error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+	bytes, err := io.ReadAll(file)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(bytes, &players)
```

Comment:
`io.ReadAll` + `json.Marhsal` works (and is how I'd mostly expect to see this code), but has the same issue we discussed on `go-cat` about reading the whole file into memory - can you find a way to avoid doing so?

Reply:
I found documentation about using `decoder.More()`that decode objects one by one , but we still going to create a slice (of players) without pre-allocated capacity, and append elements to it. Is it more efficient than `io.ReadAll`